plugins/Diagnostics/Diagnostic/DatabaseAbilitiesCheck.php

```diff
@@ -82,6 +93,25 @@ public function execute()
             );
         }
 
-        return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_WARNING, $comment));
+        return new DiagnosticResultItem(DiagnosticResult::STATUS_WARNING, $comment);
+    }
+
+    protected function checkTemporaryTables()
+    {
+        $status = DiagnosticResult::STATUS_OK;
+        $comment = 'CREATE TEMPORARY TABLES';
+
+        try {
+            Db::exec("CREATE TEMPORARY TABLE `piwik_test_table_temp` (
```

Comment:
btw the issue was not just about temporary tables probably (which probably works most of the times but also whether we can create temporary table, change the transaction status level, then do a write to temporary table, and several reads from it succeed in case there's a load balancer with multiple readers). For now that check be fine but I think the original issue was a bit more complicated where temporary tables itself weren't an issue.

Reply:
it's not important though for now and can schedule this also for 4.1

Reply:
fyi @sgiehl

Reply:
I've added a check that writing and reading from the temporary table works.
Not sure if we should add checking if transactions work correctly to a new check, as it should be unrelated to temporary tables at all?

Reply:
We will need to check transaction status level not transactions itself. This can be done in a different check though.

Reply:
> check transaction status level

actually I'm not sure how to do that :man_shrugging:

Reply:
I think a check could look like this:

```php
$level = StaticContainer::get(TransactionLevel::class);
if (!$level->setUncommitted()) {
    // warning: Changing transaction isolation level not supported. Archiving will still work but it may be slower and it is recommended to change for example the `binlog_format` to `row` if possible.
} else {
$level->restorePreviousStatus();
}

Reply:
added an additional check for that now